File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r, mini = 0, len(nums)-1, 1001
        while(l<r):
            m=l+(r-l)//2
            mini = min(mini, nums[m])
            if nums[m]>nums[r]:
                l=m+1
            elif nums[m]<nums[r]:
                r=m
        return min(mini, nums[l])

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mini = 101, maxi = -1, n=prices.size();
        for(int i=0;i<n;i++){
            mini = min(mini, prices[i]);
            maxi = max(maxi, prices[i]-mini);
        }
        return maxi;
    }
};
